background.js

```diff
@@ -120,14 +117,28 @@ ipcMain.on('k8s-restart', () => {
     k8smanager.stop()
       .then(() => {
         // The desired Kubernetes version might have changed
-        k8smanager = K8s.factory(cfg.kubernetes);
+        k8smanager = newK8sManager(cfg.kubernetes);
       })
-      .then(() => { k8smanager.start() })
+      .then(() => { return k8smanager.start() })
       .then(() => {
+        try {
+          event.reply('k8s-check-state', k8smanager.state);
+        } catch (err) {
+          console.log(err);
+        }
       }, startfailed);
   }
 })
 
 function startfailed(code) {
   dialog.showErrorBox("Error Starting Kuberentes", "Kubernetes was unable to start with the following exit code: " + code);
+}
+
+function newK8sManager(cfg) {
+  let mgr = K8s.factory(cfg);
+  mgr.on("state-changed", (state) => {
+    tray.k8sStateChanged(state);
+  })
```

Comment:
```suggestion
  });
```
Nit so pointless I felt I should make it trivial to accept ;)

Reply:
I'm making my own change because there should be a `;` at the end of the next code line.. the return statement. Fixing that, too.